src/account.ts

```diff
@@ -824,109 +811,139 @@ export default class EthereumAccount {
         this.subBalance(amount)
         this.account.payoutAmount = this.account.payoutAmount.plus(amount)
       } catch (err) {
-          // Balance update likely dropped below the minimum, so throw an internal error
+        // Balance update likely dropped below the minimum, so throw an internal error
         this.master._log.trace(`Failed to fulfill response to PREPARE: ${err.message}`)
         throw new IlpPacket.Errors.InternalError(err.message)
       }
     }
 
-    if (attemptSettle) {
-      this.outgoingSettlements.runExclusive(() => this.attemptSettle())
-        .catch(err => {
-          this.master._log.trace(`Error queueing an outgoing settlement: ${err.message}`)
-        })
+    // Attempt to settle on fulfills and* T04s (to resolve stalemates)
+    const shouldSettle = isFulfill || (isReject && responsePacket.data.code === 'T04')
+    if (shouldSettle) {
+      this.attemptSettle().catch((err: Error) => {
+        this.master._log.trace(`Error queueing an outgoing settlement: ${err.message}`)
+      })
     }
   }
 
-  private startChannelWatcher (): void {
-    const interval = this.master._channelWatcherInterval.toNumber()
-    const timer: NodeJS.Timer = setInterval(async () => {
-      try {
+  private startChannelWatcher () {
+    const timer: NodeJS.Timeout = setInterval(
+      () => this.queue.synchronize(TaskPriority.ChannelWatcher, async () => {
         const claim = this.account.bestIncomingClaim
-        if (!claim) return
+        if (!claim) {
+          // No claim is linked: stop the channel watcher
+          this.watcher = null
+          return clearInterval(timer)
+        }
 
-        const channel = await this.updateChannel(claim.channelId)
+        const channel = await this.fetchChannel(claim.channelId)
+        if (!channel) {
+          // Channel is closed: stop the channel watcher
+          this.watcher = null
+          return clearInterval(timer)
+        }
 
-        if (channel && isSettling(channel)) {
-          await this.claimIfProfitable(true)
+        if (isSettling(channel)) {
+          this.claimIfProfitable(true).catch((err: Error) => {
+            this.master._log.trace(`Error attempting to claim channel: ${err.message}`)
+          })
         }
-      } catch (err) {
-        this.master._log.trace(`Error while running channel watcher: ${err.message}`)
-      }
-    }, interval)
+      }),
+      this.master._channelWatcherInterval.toNumber()
+    )
 
-    // Check if we're in a Node.js environment
+    // Don't let the timer prevent existing the process if we're in Node.js
```

Comment:
Should it make sure it's done claiming before exiting?

Reply:
Hmmm...I'll get rid of that. It'd be weird to have the process exit while the channel watcher is still running -- that's probably unexpected behavior.

(This also suggests unref is an anti-pattern: https://www.bennadel.com/blog/3301-calling-timeout-unref-with-settimeout-does-not-appear-to-be-a-best-practice-in-node-js.htm).